IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometryResource/Functions/IfcGradient/Expression.txt

```diff
@@ -1,14 +1,10 @@
 (GradientCurve : IfcGradientCurve, Position : IfcCurveMeasureSelect) : IfcLengthMeasure;
-  
-  LOCAL
-    DistanceAlong : IfcLengthMeasure := 0;
-    Elevation : IfcLengthMeasure := 0;
-    SegmentPlacement : IfcAxis2Placement2D := Segment.Placement\IfcAxis2Placement2D;
-  END_LOCAL;
-  
+    
   REPEAT i := 1 TO SIZEOF(GradientCurve.Segments);//GradientCurve.BaseCurve\IfcCompositeCurve.Segments;
-    Segment : IfcCurveSegment := GradientCurve.Segments[i];//BaseCurve\IfcCompositeCurve.Segments[i];  NextSegment : IfcCurveSegment :=     GradientCurve.Segments[i + 1];//BaseCurve\IfcCompositeCurve.Segments[i + 1];
+    Segment : IfcCurveSegment := GradientCurve.Segments[i];
+    NextSegment : IfcCurveSegment := GradientCurve.Segments[i + 1];
```

Comment:
In this case, the loop should iterate to `SIZEOF( ..Segments ) - 1` to not overflow the array boundaries.

Reply:
Oh yes, true. Old habbits.